### PR TITLE
WX-1595 Fix batch workflow options override

### PR DESCRIPTION
**Pull request by aednichols**
### Description

Replaces https://github.com/broadinstitute/cromwell/pull/7460
Fixes https://github.com/broadinstitute/cromwell/issues/7459

#### `CHANGELOG.md`
 - [ ] I updated `CHANGELOG.md` in this PR
 - [ ] I assert that this change shouldn't be included in `CHANGELOG.md` because it doesn't impact community users

#### Terra Release Notes
 - [ ] I added a suggested release notes entry in this Jira ticket
 - [ ] I assert that this change doesn't need Jira release notes because it doesn't impact Terra users

### CODE REVIEW

**Comment by aednichols** [comment]
Fixes https://github.com/broadinstitute/cromwell/issues/7459

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ The `IX_WORKFLOW_STORE_ENTRY_WS` index is removed from `WORKFLOW_STORE_ENTRY`.
 
 The index had low cardinality and workflow pickup is faster without it. Migration time depends on workflow store size, but should be very fast for most installations. Terminal workflows are removed from the workflow store, so only running workflows contribute to the cost.
 
+### Bug Fixes
+
+* Fixed `google_project` and `google_compute_service_account` workflow options not taking effect when using GCP Batch backend
```
Comment (AlexITC) [inline CHANGELOG.md:37]: There were a few fixes included in the `GCP Batch` section, should we move this?

Comment (aednichols) [reply CHANGELOG.md:37]: Good point. Will fix!

**Comment by aednichols** [comment]
Will admin-merge once CI completes due to two thumbs on https://github.com/broadinstitute/cromwell/pull/7460

**Pull request merged**